src/lib/Preview.js

```diff
@@ -879,7 +897,9 @@ class Preview extends EventEmitter {
             });
 
             // If there wasn't an error, use Events API to log a preview
-            this.logPreviewEvent(this.file.id, this.options);
+            if (!this.options.skipServerUpdate) {
+                this.logPreviewEvent(this.file.id, this.options);
+            }
```

Comment:
Guys, agree, let the request fail it was offline. 
@priyajeet Thanks for good catch about `this.prefetchNextFiles()`. I've added the guard.